### PR TITLE
[repo] Package validation tweaks

### PR DESCRIPTION
**Pull request by CodeBlanch**
Follow-up to #4767

## Changes

* Removed `ApiCompatBaseline.txt` from SDK as the new analyzer ignores nullable attributes
* Moved the `PackageValidationBaselineVersion` definition next to the `OTelLatestStableVer` definition
* Added `EnablePackageValidation` property so it is easy to turn on/test package validation from IDE

### CODE REVIEW

**Review by utpilla — APPROVED** [review]

File: Directory.Packages.props [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
     <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
-  </PropertyGroup>
-
-  <PropertyGroup>
     <OTelLatestStableVer>1.6.0</OTelLatestStableVer>
   </PropertyGroup>
 
+  <Target Name="PreparePackageValidation" BeforeTargets="Pack">
+    <PropertyGroup>
+      <EnablePackageValidation Condition="'$(EnablePackageValidation)' == ''">false</EnablePackageValidation>
```
Comment (utpilla) [inline Directory.Packages.props:9]: Is this needed?

Comment (utpilla) [reply Directory.Packages.props:9]: Just noticed that this is answered in the PR description.

File: Directory.Packages.props [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
-  <PropertyGroup>
     <OTelLatestStableVer>1.6.0</OTelLatestStableVer>
   </PropertyGroup>
 
+  <Target Name="PreparePackageValidation" BeforeTargets="Restore">
+    <PropertyGroup>
+      <EnablePackageValidation Condition="'$(EnablePackageValidation)' == ''">true</EnablePackageValidation>
+      <PackageValidationBaselineVersion Condition="'$(MinVerTagPrefix)' == 'core-'">$(OTelLatestStableVer)</PackageValidationBaselineVersion>
+    </PropertyGroup>
+  </Target>
```
Comment (ViktorHofer) [inline Directory.Packages.props:12]: It's an anti-pattern to have a target in a props file.

File: Directory.Packages.props [hunk 3]
```diff
@@ -1,12 +1,16 @@
...
     <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
-  </PropertyGroup>
-
-  <PropertyGroup>
     <OTelLatestStableVer>1.6.0</OTelLatestStableVer>
   </PropertyGroup>
 
+  <Target Name="PreparePackageValidation" BeforeTargets="Restore">
+    <PropertyGroup>
+      <EnablePackageValidation Condition="'$(EnablePackageValidation)' == ''">true</EnablePackageValidation>
```
Comment (ViktorHofer) [inline Directory.Packages.props:9]: Now that this property defaults to true, you can remove it from the YML here: https://github.com/open-telemetry/opentelemetry-dotnet/blob/d76542e9daa40051a9e5408bc7a849a6c3974684/.github/workflows/package-validation.yml#L22

File: Directory.Packages.props (hunk 2 above)
Comment (CodeBlanch) [reply Directory.Packages.props:12]: Fixed on #4927

File: Directory.Packages.props (hunk 3 above)
Comment (CodeBlanch) [reply Directory.Packages.props:9]: I actually intended this to be `false`! Must have set it to `true` when I switched to the `Restore` phase. Woops! Fixed on #4927

**Pull request merged**